src/app/(shared-nav)/courses/page.tsx

```diff
@@ -0,0 +1,101 @@
+"use client"
+import { languages } from "@/app/enums/language.enum";
+import { levels } from "@/app/enums/level.enum";
+import { CoursCard } from "@/components/coursCard";
+import { coursDetails } from "@/types/coursDetails.type";
+import { ChangeEvent, useEffect, useState } from "react";
+import { IoSearch } from "react-icons/io5";
+
+export default function Courses(){
+    const [searchText, setSearchText] = useState<String>("");
+    const [courses, setCourses] = useState<coursDetails[]>(DATA);
+    const handleSearch = (event: ChangeEvent<HTMLInputElement>)=>{
+        setSearchText(event.target.value);
+    }
+    useEffect(()=>{
+        if(searchText !== "")
+            setCourses(DATA.filter(cours=> cours.title.includes(searchText as string)))
+        else if([null, undefined, ""].includes(searchText as (string | null | undefined))){
```

Comment:
there is a better way of doing this 

useEffect(() => {
  if (searchText) {
    setCourses(DATA.filter(course => course.title.includes(searchText)));
  } else {
    setCourses(DATA);
  }
}, [searchText]);